Source/Eto.Gtk/Forms/Controls/GridHandler.cs

```diff
@@ -238,7 +268,27 @@ public int NumberOfColumns
 
 		public int GetColumnOfItem(Gtk.TreeViewColumn item)
 		{
-			return Widget.Columns.Select(r => r.Handler as GridColumnHandler).Select(r => r.Control).ToList().IndexOf(item);
+			for (int i = 0; i < Tree.Columns.Length; i++)
+				if (Tree.GetColumn(i) == item && i < Widget.Columns.Count)
+					return i;
+
+			return -1;
+		}
+
```

Comment:
The original code did not work 100%.
It sometimes happened that the cast was unsuccesfull.
It was very irregular. Sometimes it worked, sometimes not.

I also tested the code below but it didn't work.
It seems that when casting `Widget.Columns[someColum,].Handler` to `GridColumnHandler` that sometimes (not every time) some data gets lost. One of the things that can get lost is the column title (it changes to `null`).
```C#
for (int i = 0; i < Widget.Columns.Count; i++)
{
    if ((Widget.Columns[i].Handler as GridColumnHandler).Control == item)
        return i;
}
return -1;
```
I tried to explain it the best I could but English is not my native language.